content/inside-rust/stage0-redesign/index.md

```diff
@@ -0,0 +1,70 @@
+authors = ["Jieyou Xu"]
+
+[extra]
+team = "Bootstrap"
+team_url = "https://www.rust-lang.org/governance/teams/infra#team-bootstrap"
++++
+
+# Redesigning stage 0 std
+
+## Summary
+
+We are reworking how the stage 0 bootstrap sequence works (the sequence used to build a stage 1 compiler).
+
+- Before: a stage 0 (beta by default) compiler is used to build the in-tree std, which in turn is used to build the stage 1 compiler.
+- After: a stage 0 (beta by default) compiler and a precompiled stage 0 std is instead used to build the stage 1 compiler.
+
+![Difference in stage 0 bootstrap sequence](stage0-redesign-diff.svg)
+
+Notably, this means that after [redesign stage 0 std #119899](https://github.com/rust-lang/rust/pull/119899) PR lands:
+
+- `./x {build,test,check} library --stage 0` becomes no-op, as stage 0 std is no longer the built in-tree std, and the minimum supported stage to build std is now `1`.
+    - Consequently, default (test, check, bench) stage values in the library profile are no longer `0`, but instead defaults to `1`.
+- Some additional `cfg(bootstrap)` usages may be needed in the compiler sources for dogfooding unstable library features.
+
+If you have questions, or run into bugs/issues related to the stage 0 std redesign, please discuss them in the [stage 0 std redesign support thread][support-thread] or open a new issue.
+
+[support-thread]: https://rust-lang.zulipchat.com/#narrow/channel/326414-t-infra.2Fbootstrap/topic/Stage.200.20std.20redesign.20support.20thread/with/515096924
+
+## Comparison of common invocations
+
+For `profile = "library"`:
+
+| Invocation                                 | Before stage 0 std redesign                                 | After stage 0 std redesign                               |
+|--------------------------------------------|-------------------------------------------------------------|----------------------------------------------------------|
+| `./x {check,build,test} library`           | Checks/builds/tests in-tree library                         | Check/build/tests stage 1 library                        |
+| `./x {check,build,test} library --stage 0` | Checks/builds/tests in-tree library                         | No-op in case of build, checks/tests precompiled library |
+| `./x {check,build,test} library --stage 1` | Builds in-tree library, checks/builds/tests stage 1 library | Check/build/tests stage 1 library                        |
+
+For `profile = "compiler"`:
+
+| Invocation                                 | Before stage 0 std redesign                                 | After stage 0 std redesign                               |
+|--------------------------------------------|-------------------------------------------------------------|----------------------------------------------------------|
+| `./x check library`                        | Checks in-tree library                                      | Check/build/tests stage 1 library                        |
```

Comment:
```suggestion
| `./x check library`                        | Checks in-tree library                                      | Check stage 1 library                        |
```
or does `x check library` actually build and test it?

Reply:
No, that's a mistake, good catch

Reply:
I dropped these "specific" invocations in (round 4). I found them more noise than signal.